Runtime/Interaction/BaseInteractionSimulator.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using System.Linq;
+using Innoactive.Creator.BasicInteraction;
+using Innoactive.Creator.Core.Utils;
+using UnityEngine.SceneManagement;
+
+namespace Innoactive.Creator.BasicInteraction
+{
+    public abstract class BaseInteractionSimulator
+    {
+        private static BaseInteractionSimulator instance;
+        public static BaseInteractionSimulator Instance
+        {
+            get
+            {
+                if (instance == null)
+                {
+                    Type type = ReflectionUtils
+                        .GetConcreteImplementationsOf(typeof(BaseInteractionSimulator))
+                        .FirstOrDefault(t => t != typeof(BaseInteractionSimulatorDummy));
+
+                    if (type == null)
+                    {
+                        type = typeof(BaseInteractionSimulatorDummy);
+                    }
+
+                    instance = (BaseInteractionSimulator)ReflectionUtils.CreateInstanceOfType(type);
+                    SceneManager.sceneUnloaded += OnSceneLoad;
+                }
+
+                return instance;
+            }
+        }
+
+        private static void OnSceneLoad(Scene scene)
+        {
+            instance = null;
+            SceneManager.sceneUnloaded -= OnSceneLoad;
+        }
+
+        /// <summary>
+        /// Simulates touching the given object. To un touch it again, run the specific method.
```

Comment:
Either this suggestion or all are redundant.
```suggestion
        /// Simulates touching the given object.
```